vcd/resource_vcd_external_network.go

```diff
@@ -0,0 +1,334 @@
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: singleIP(),
+									},
+								},
+							},
+						},
+					},
+				},
+				Description: "",
+			},
+			"vim_port_group": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: false,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"vim_server": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"mo_ref": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"vim_object_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"fence_mode": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "",
+			},
+			"retain_net_info_across_deployments": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "",
+			},
+			"parent_network": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "",
+			},
+		},
+	}
+}
+
+// Creates a new external network from a resource definition
+func resourceVcdExternalNetworkCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] external network creation initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	// external network creation is accessible only in administrator API part
+	_, err := vcdClient.GetAdminOrgFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrg, err)
+	}
+
+	params, err := getExternalNetworkInput(d, vcdClient)
+	if err != nil {
+		return err
+	}
+
+	task, err := govcd.CreateExternalNetwork(vcdClient.VCDClient, params)
+	if err != nil {
+		log.Printf("[DEBUG] Error creating external network: %#v", err)
+		return fmt.Errorf("error creating external network: %#v", err)
+	}
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		log.Printf("[DEBUG] Error waiting for external network to finish: %#v", err)
+		return fmt.Errorf("error waiting for external network to finish: %#v", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+	log.Printf("[TRACE] external network created: %#v", task)
+	return resourceVcdExternalNetworkRead(d, meta)
+}
+
+// Fetches information about an existing external network for a data definition
+func resourceVcdExternalNetworkRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] external network read initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	externalNetworkRef, err := govcd.GetExternalNetworkByName(vcdClient.VCDClient, d.Id())
+	if err != nil || *externalNetworkRef == (types.ExternalNetworkReference{}) {
+		log.Printf("[DEBUG] Unable to find external network. Removing from tfstate")
+		d.SetId("")
+		return nil
+	}
+
+	externalNetwork := govcd.NewExternalNetwork(&vcdClient.VCDClient.Client)
```

Comment:
What is idea behind additional refresh after fetching external network?

Reply:
This is no longer needed with the changes to `govcd`.